### PR TITLE
[clang] Function Multi Versioning supports IFunc lowerings on Darwin platforms

### PR DESCRIPTION
**Pull request by jroelofs**

### CODE REVIEW

**Comment by tahonermann** [comment]
I recommend extending testing via additional `RUN` lines to all of the following tests:
- `clang/test/CodeGen/attr-cpuspecific.c`
- `clang/test/CodeGen/attr-target-clones.c`
- `clang/test/CodeGen/attr-target-mv-func-ptrs.c`
- `clang/test/CodeGen/attr-target-mv-va-args.c` (already done)
- `clang/test/CodeGen/attr-target-mv.c`
- `clang/test/CodeGenCXX/attr-cpuspecific.cpp`
- `clang/test/CodeGenCXX/attr-target-mv-diff-ns.cpp`
- `clang/test/CodeGenCXX/attr-target-mv-inalloca.cpp`
- `clang/test/CodeGenCXX/attr-target-mv-member-funcs.cpp`
- `clang/test/CodeGenCXX/attr-target-mv-modules.cpp`
- `clang/test/CodeGenCXX/attr-target-mv-out-of-line-defs.cpp`
- `clang/test/CodeGenCXX/attr-target-mv-overloads.cpp`
- `clang/test/CodeGenCXX/attr-target-clones.cpp`

Note that, for many of the above tests, changes planned in #71706 will impact the expected behavior.

**Review by MaskRay — APPROVED** [review]
This PR just updates `supportsIFunc` and extends clang tests. LGTM

**Review by ilinpv — APPROVED** [review]
LGTM!

File: clang/test/CodeGenCXX/attr-cpuspecific.cpp [hunk 1]
```diff
@@ -1,4 +1,5 @@
 // RUN: %clang_cc1 -triple x86_64-linux-gnu -emit-llvm -o - %s | FileCheck %s --check-prefix=LINUX
+// RUN: %clang_cc1 -triple x86_64-apple-macosx -emit-llvm -o - %s | FileCheck %s --check-prefix=LINUX
```
Comment (MaskRay) [inline clang/test/CodeGenCXX/attr-cpuspecific.cpp:2]: Is "macosx" a deprecated OS name?

**Pull request merged**